test/view_component/integration_test.rb

```diff
@@ -583,4 +583,19 @@ def test_sets_the_compiler_mode_in_development_mode
       assert_equal ViewComponent::Compiler::DEVELOPMENT_MODE, ViewComponent::Compiler.mode
     end
   end
+
+  def test_config_options_shared_between_base_and_engine
+    Rails.application.config.view_component.yield_self do |config|
+      {
+        generate: config.generate.dup.tap { |c| c.sidecar = true },
+        preview_controller: "SomeOtherController",
+        preview_route: "/some/other/route",
+        show_previews_source: true
```

Comment:
Got a question about this - these mirror what I've introduced at `lib/view_component/engine.rb:17`. It'd be ideal that if a new config option is introduced, it only needs to be updated in one place. Intuitively, I'd say we do that by defining a constant `ViewComponent::Engine::CONFIG_OPTIONS_SHARED_WITH_BASE`, which is that array I'm using to mirror the config options in `view_component.set_configs`. I'm not so sure about creating a constant purely for the sake of testing, though. What do folks think?

Reply:
Talked with @ollie-nye about this, and we weren't really able to come to a convincing conclusion within the way things work at present. So I'm thinking maybe we should have a `Config` object which `ViewComponent::Base` and Rails config both point to. That way, the only test we need is to ensure that `ViewComponent::Base.config.<option>` mirrors `Rails.application.config.view_component.<option>` - then we don't need to worry about updating this test as new options are introduced.

Reply:
I should add that if this is a good idea, I'll do it in a separate PR. I can imagine there's a lot that it may come to affect - we'll have to deprecate all the `matter_accessor`s on `ViewComponent::Base`.

Reply:
@boardfish I think we should only have them implemented once. Would it make sense to make that change first?

Reply:
Now that I think about it, I guess we could achieve this without needing to deprecate the `mattr_accessor`s as long as they point to this new config object too.

I'll continue on with that in mind - one implementation of the config that is used across all current accessors.

Reply:
Been busy lately, so this has admittedly fallen by the wayside. I should be able to resume work on it early next week.